cmake/kokkos_compiler_id.cmake

```diff
@@ -107,6 +107,11 @@ IF(KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
   IF (INTERNAL_HAVE_INTEL_COMPILER) #not actually Clang
     SET(KOKKOS_CLANG_IS_INTEL TRUE)
     SET(KOKKOS_CXX_COMPILER_ID IntelLLVM CACHE STRING INTERNAL FORCE)
+    EXECUTE_PROCESS(COMMAND ${CMAKE_CXX_COMPILER} --version
+                  OUTPUT_VARIABLE INTERNAL_CXX_COMPILER_VERSION
+                  OUTPUT_STRIP_TRAILING_WHITESPACE)
+    STRING(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]+"
+           KOKKOS_CXX_COMPILER_VERSION ${INTERNAL_CXX_COMPILER_VERSION})
```

Comment:
Did you mean to do this unconditionally (regardless of the CMake version)?

Reply:
It looks like the `IntelLLVM` `CMAKE_<LANG>_COMPILER_ID` was added in 3.20.

Reply:
Should we guard that block
https://github.com/kokkos/kokkos/blob/a0ffec1857c2195bf5340a17dfd3d87a45d45eee/cmake/kokkos_compiler_id.cmake#L102-L115
then

Reply:
We already guard this block since the compiler id reports as `IntelLLVM` and not `Clang` for `CMake` >=3.20. We could add a remark, though, if you prefer.